streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -349,6 +359,57 @@ public String queryableStoreName() {
         return toStream().selectKey(mapper);
     }
 
+    @Override
+    public KTable<K, V> suppress(final Suppressed<K, V> suppressed) {
+        final SuppressedImpl<K, V> suppressImpl = (SuppressedImpl<K, V>) suppressed;
+        final String name = builder.newProcessorName(SUPPRESS_NAME);
+
+        final ProcessorSupplier<K, Change<V>> suppressionSupplier =
+            () -> new KTableSuppressProcessor<>(buildSuppress(suppressImpl));
+
+        final ProcessorParameters<K, Change<V>> processorParameters = new ProcessorParameters<>(
+            suppressionSupplier,
+            name
+        );
+
+        final ProcessorGraphNode<K, Change<V>> node = new ProcessorGraphNode<>(name, processorParameters, false);
+
+        builder.addGraphNode(this.streamsGraphNode, node);
+
+        return new KTableImpl<K, S, V>(
+            builder,
+            name,
+            suppressionSupplier,
+            keySerde,
+            valSerde,
+            Collections.singleton(this.name),
+            null,
+            false,
+            node
+        );
+    }
+
+    @SuppressWarnings("unchecked") // see the comments on the casts
+    private SuppressedImpl<K, V> buildSuppress(final SuppressedImpl<K, V> suppress) {
+        if (suppress.isFinalResultsSuppression()) {
+            // This throws an exception if the computation is not windowed
+            final long grace = findAndVerifyWindowGrace(streamsGraphNode);
```

Comment:
I figured that "verify" means "orThrow".

The comment is part of a series that explains why the casts in this method are safe.

I think that it would be very difficult to understand later why it's ok to cast `K` to `Windowed`, and then why we would cast the resultant suppression back to `K`.

Reply:
Ack -- was just a thought.